app/controllers/idv/cancellations_controller.rb

```diff
@@ -33,6 +35,31 @@ def destroy
 
     private
 
+    def barcode_step?
+      params[:step] == 'barcode'
+    end
+
+    def enrollment
+      InPersonEnrollment.where(user_id: current_user.id).first
+    end
+
+    def extra_analytics_attributes
+      extra = {}
+      if barcode_step? && enrollment
+        extra.merge!(
+          cancelled_enrollment: false,
+          enrollment_code: enrollment.enrollment_code,
+          enrollment_id: enrollment.id,
+          service_provider: decorated_session.sp_name || APP_NAME,
```

Comment:
Do we need to include this, or can we rely on it being provided in the default event attributes?

https://github.com/18F/identity-idp/blob/f226bddb2477ef19c2b04ccd32f056c3a5a57dad/app/services/analytics.rb#L73

```suggestion
```

Reply:
you're right, we do get that in the default - i'll remove this addition